tools/cli/util.go

```diff
@@ -714,7 +714,7 @@ func parseTime(timeStr string, defaultValue int64) int64 {
 func parseTimeRange(timeRange string) (time.Time, error) {
 	match, err := regexp.MatchString(defaultDateTimeRangeShortRE, timeRange)
 	if !match { // fallback on to check if it's of longer notation
-		match, err = regexp.MatchString(defaultDateTimeRangeLongRE, timeRange)
+		_, err = regexp.MatchString(defaultDateTimeRangeLongRE, timeRange)
 	}
```

Comment:
not necessarily relevant as this doesn't change behavior, but:  what if the second doesn't match?  looks like `regexp.MatchString` will only error if the regex fails to compile.

more generally, it seems pretty obvious that we should only be compiling these once, rather than on every call :\  think that's worth doing here, or later?

Reply:
Good observation. Few problems here:
- regexes are not compiled as you mentioned (although not very impactful as those are used only on CLI commands)
- these matches are not used anywhere. Error simply check whether they compile.
- Match groups in those regexes are not utilized, we parse things manually below

I think I will address this in a separate PR. Also checked that there exists a library for such functionality: https://github.com/karrick/tparse
What is opinion on taking this dependency? Or just simply fixing is enough?